Add ebpf.io (#36)
zoidbergwill/awesome-ebpf: readme.md

BPF, as in _Berkeley Packet Filter_, is an in-kernel virtual machine running programs passed from user space. Initially implemented on BSD, then Linux, the (now legacy) "classic BPF" or cBPF machine would be used with tools like tcpdump for filtering packets in the kernel to avoid useless copies to user space. More recently, the BPF infrastructure in Linux has been completely reworked and gave life to the "extended BPF", or eBPF, which gained new features (safety and termination checks, JIT-compiling for programs, persistent maps, a standard library, hardware offload support, etc.) and is now used for many tasks. Processing packets at a very low level (XDP), tracing and monitoring events on the system, or enforcing access control over cgroups are but a few examples to which eBPF brings performance, programmability and flexibility.

> Recently Cilium launched a great website about eBPF called [epf.io](https://ebpf.io/). It serves a similar purpose to this list, with a summary of eBPF and links to related [projects](https://ebpf.io/projects).

Reviewer: Maybe:
```suggestion
Recently [Cilium](https://cilium.io) launched a great website about eBPF called [ebpf.io](https://ebpf.io/). It serves a similar purpose to this list, with [an introduction to eBPF](https://ebpf.io/what-is-ebpf) and links to [related projects](https://ebpf.io/projects).
```

### Cilium Guide

> - [Cilum's New BPF and XDP Reference Guide](https://ebpf.io/) - General documentation about features of eBPF, with a lot of pretty diagrams, and links to a bunch of related [projects](https://ebpf.io/projects).

Author: The Cilium section is the first section of links, but you're right maybe it belongs above the Table of Contents!
